src/OrchardCore.Modules/OrchardCore.Autoroute/Models/AutoroutePartExtensions.cs

```diff
@@ -14,16 +15,26 @@ public static IEnumerable<ValidationResult> ValidatePathFieldValue(this Autorout
                 yield return new ValidationResult(S["Your permalink can't be set to the homepage, please use the homepage option instead."], new[] { nameof(autoroute.Path) });
             }
 
-            if (autoroute.Path?.IndexOfAny(AutoroutePart.InvalidCharactersForPath) > -1 || autoroute.Path?.IndexOf(' ') > -1 || autoroute.Path?.IndexOf("//") > -1)
+            if (HasInvalidCharacters(autoroute.Path))
             {
                 var invalidCharactersForMessage = string.Join(", ", AutoroutePart.InvalidCharactersForPath.Select(c => $"\"{c}\""));
-                yield return new ValidationResult(S["Please do not use any of the following characters in your permalink: {0}. No spaces, or consecutive slashes, are allowed (please use dashes or underscores instead).", invalidCharactersForMessage], new[] { nameof(autoroute.Path) });
+                const string errorMessage = "Please do not use any of the following characters in your permalink: {0}. No spaces, or consecutive slashes, are allowed (please use dashes or underscores instead).";
```

Comment:
doing this means it will no longer be localized. please do that as it was before.

Reply:
@deanmarcussen I pull this error message to a variable for better code formatting.
However, I think with this code `S[errorMessage, invalidCharactersForMessage]` it should be get translated as it was before.

Please correct me if I am wrong for this.

Reply:
@deanmarcussen  Here is my test result with `th-TH` culture.
The translation still works.

![image](https://user-images.githubusercontent.com/2938310/127264267-7812552d-c2f2-4c28-a708-17031f657d50.png)

Here is a content in th-TH.po.
```
msgid "Please do not use any of the following characters in your permalink: {0}. No spaces, or consecutive slashes, are allowed (please use dashes or underscores instead)."
msgstr "กรุณาอย่าใช้ตัวอักษรดังต่อไปนี้ใน permalink : {0}. ห้ามใส่ช่องว่าง, consecutive slashes (กรุณาใช้ dash หรือ underscores แทน)"
```

Reply:
Please revert the change to the localization. It will not work as the poextractor relies on the string being provided in the S accessor.

Reply:
@deanmarcussen I have already reverted the localization message, please check all changes to this PR.
https://github.com/OrchardCMS/OrchardCore/pull/9985/files

I think is ready to be merged.
Thank you so much.